### ISSUE TITLE
ArrayInput does not support helperText prop

### ISSUE THREAD

**Issue by netdown** [marmelab/react-admin#4227]
**What you were expecting:**
As ArrayInput should accept the common input props, helperText should be working, too.

**What happened instead:**
HelperText prop is not handled in the input, therefore does not appear.

**Related code:**
I'd love to create a pull request, however, I am not sure how the related files are compiled. Making the prop work would only need a few lines of code into ArrayInput.tsx:

```
import {  InputLabel, FormControl, FormHelperText } from '@material-ui/core';
...
<InputLabel ... />
<FormHelperText>{helperText}</FormHelperText>
```
Besides this, adding helperText const to the props and the proptypes.

**Environment**

* React-admin version: 3.1.1

**Comment by ThieryMichel** [comment]
Thanks for the issue

**Comment by fzaninotto** [comment]
I don't think it makes sense. An ArrayInput isn't a regular input, it takes a form iterator as child, and displays a list of inputs (each of which can have their own helperText). 

Where would you want to display the helper text? Can you post an example screenshot?

**Comment by netdown** [comment]
Sure, attached.
![ex](https://user-images.githubusercontent.com/4265403/71730424-62e33c80-2e42-11ea-8785-d2aef9ed2bcf.png)

It does make sense, because the inputs' helper texts give information about the exact inputs. However, the list itself may also require an explanation. For example, on the screenshot I let the user define custom states (currently only a name, but does not matter) for recorded bookings. I would like to display some helper text to explicitly tell the user what is this list and what is it for. Clearly, it would look strange if this text displayed under each input.

**Comment by fzaninotto** [comment]
OK, understood. I'll mark this issue as an enhancement request. This means that if someone opens a PR to add this feature, we'll merge it. But if nobody provides the code for it within a few months, as we don't need it ourselves, we'll close this issue.

Note that you can still do it in userland by writing your own FormIterator.

**Comment by WiXSL** [comment]
This can be closed

**Comment by djhi** [comment]
Fixed by #6062
